src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+    #[MethodInput=WriteInput,rpc_name="file/write",result=write_result,set_result=set_write_result]
+    fn write(&self, path:Path, contents:String) -> ();
+
+    /// Watches the specified path.
+    #[MethodInput=CreateWatchInput,rpc_name="file/createWatch",result=create_watch_result,set_result=set_create_watch_result]
+    fn create_watch(&self, path:Path) -> Uuid;
+
+    /// Delete the specified watcher.
+    #[MethodInput=DeleteWatchInput,rpc_name="file/deleteWatch",result=delete_watch_result,
+    set_result=set_delete_watch_result]
+    fn delete_watch(&self, watch_id:Uuid) -> ();
+}
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::FileKind::RegularFile;
+
+    use json_rpc::messages::Message;
+    use json_rpc::messages::RequestMessage;
+    use json_rpc::test_util::transport::mock::MockTransport;
+    use serde_json::json;
+    use serde_json::Value;
+    use std::future::Future;
+    use utils::test::poll_future_output;
+    use utils::test::poll_stream_output;
+    use futures::task::LocalSpawnExt;
+
+    struct Fixture {
+        transport : MockTransport,
+        client    : Client,
+        executor  : futures::executor::LocalPool,
+    }
+
+    fn setup_fm() -> Fixture {
+        let transport = MockTransport::new();
+        let client    = Client::new(transport.clone());
+        let executor  = futures::executor::LocalPool::new();
+        executor.spawner().spawn_local(client.runner()).unwrap();
+        Fixture {transport,client,executor}
+    }
+
+    #[test]
+    fn test_notification() {
+        let mut fixture = setup_fm();
+        let mut events  = Box::pin(fixture.client.events());
+        assert!(poll_stream_output(&mut events).is_none());
+
+        let expected_notification = FilesystemEvent {
+            path : Path::new("./Main.luna"),
+            kind : FilesystemEventKind::Modified,
+        };
+        let notification_text = r#"{
+            "jsonrpc": "2.0",
+            "method": "filesystemEvent",
+            "params": {"path" : "./Main.luna", "kind" : "Modified"}
+        }"#;
+        fixture.transport.mock_peer_message_text(notification_text);
+        assert!(poll_stream_output(&mut events).is_none());
+
+        fixture.executor.run_until_stalled();
+
+        let event = poll_stream_output(&mut events);
+        if let Some(Event::Notification(n)) = event {
+            assert_eq!(n, Notification::FilesystemEvent(expected_notification));
+        } else {
+            panic!("expected notification event");
+        }
+    }
+
+    /// Tests making a request using file manager:
+    /// * creates FM client and uses `make_request` to make a request
+    /// * checks that request is made for `expected_method`
+    /// * checks that request input is `expected_input`
+    /// * mocks receiving a response from server with `result`
+    /// * checks that FM-returned Future yields `expected_output`
```

Comment:
Please use real sentences. Either make the points sentences or use semicolon at the end of the points.

Reply:
Please check how it is now.